atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/service/SplitKeyDelegatingTransactionServiceTest.java

```diff
@@ -0,0 +1,131 @@
+        verify(delegate1).get(1L);
+        verify(delegate2).get(2L);
+    }
+
+    @Test
+    public void getThrowsIfFunctionReturnsUnmappedValue() {
+        assertThatThrownBy(() -> delegatingTransactionService.get(7L))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("Could not find a transaction service for timestamp {}");
+    }
+
+    @Test
+    public void rethrowsExceptionsFromMappingFunction() {
+        RuntimeException ex = new IllegalStateException("bad");
+        TransactionService unusableService = new SplitKeyDelegatingTransactionService<>(
+                num -> {
+                    throw ex;
+                },
+                transactionServiceMap);
+
+        assertThatThrownBy(() -> unusableService.get(5L)).isEqualTo(ex);
+    }
+
+    @Test
+    public void putUnlessExistsDelegateDecidedByStartTimestamp() {
+        delegatingTransactionService.putUnlessExists(3L, 12L);
+        verify(delegate3).putUnlessExists(3L, 12L);
+    }
+
+    @Test
+    public void putUnlessExistsThrowsIfFunctionReturnsUnmappedValue() {
+        assertThatThrownBy(() -> delegatingTransactionService.putUnlessExists(4L, 12L))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("Could not find a transaction service for timestamp {}");
+    }
+
+    @Test
+    public void getMultipleDelegatesRequestsToGetMultipleOnDelegates() {
+        when(delegate1.get(eq(ImmutableList.of(1L)))).thenReturn(ImmutableMap.of(1L, 2L));
+        assertThat(delegatingTransactionService.get(ImmutableList.of(1L)))
+                .isEqualTo(ImmutableMap.of(1L, 2L));
+        verify(delegate1).get(eq(ImmutableList.of(1L)));
+    }
+
+    @Test
+    public void getMultiplePartitionsRequestsAndMergesMaps() {
+        when(delegate1.get(eq(ImmutableList.of(1L, 41L)))).thenReturn(ImmutableMap.of(1L, 8L, 41L, 48L));
+        when(delegate2.get(eq(ImmutableList.of(12L, 32L)))).thenReturn(ImmutableMap.of(12L, 28L, 32L, 38L));
+
+        assertThat(delegatingTransactionService.get(ImmutableList.of(1L, 12L, 32L, 41L)))
+                .isEqualTo(ImmutableMap.of(1L, 8L, 12L, 28L, 32L, 38L, 41L, 48L));
+        verify(delegate1).get(eq(ImmutableList.of(1L, 41L)));
+        verify(delegate2).get(eq(ImmutableList.of(12L, 32L)));
+    }
+
+    @Test
+    public void getMultipleThrowsAndDoesNotMakeRequestsIfAnyTimestampsCannotBeMapped() {
+        when(delegate1.get(eq(ImmutableList.of(1L, 41L)))).thenReturn(ImmutableMap.of(1L, 8L, 41L, 48L));
+        assertThatThrownBy(() -> delegatingTransactionService.get(ImmutableList.of(1L, 7L, 41L)))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("A batch of timestamps {} produced some transaction service keys which are"
+                        + " unknown");
+
+        // no requests made is implicit, since delegate1 is a mock
```

Comment:
`Mockito.verifyNoMoreInteractions(delegate1)` instead of the comment?

Reply:
Sure. I initially hesitated to do this since the `@After` part of the test already catches this, but would like it to be more explicit, and `verifyNoMoreInteractions` twice isn't necessarily a bad thing.